### PR TITLE
Added dotenv support and tests

### PR DESCRIPTION
**Pull request by fa993**
## Requirements

- [ x] I have read the `CODE_OF_CONDUCT.md`.
- [ x] I have updated `CHANGELOG.md` & `tools/install/deb/changelog` to reflect my changes.

## Motivation

Addresses #1035 

## Solution

```markdown
### Added
- Added dotenv crate
- Added passing env file in the command line
```

### CODE REVIEW

**Comment by Mathieu-Lala** [comment]
I opened a PR on your repo @fa993 , once your merged it in your `develop`, we can merge your `develop` in vSMTP

**Comment by fa993** [comment]
> I opened a PR on your repo @fa993 , once your merged it in your `develop`, we can merge your `develop` in vSMTP.
> [fa993#1](https://github.com/fa993/vSMTP/pull/1)

@Mathieu-Lala I merged your PR

**Comment by Mathieu-Lala** [comment]
There is just a conflict in the `Cargo.toml`, take both dotenv and the latest version of `clap`

**Comment by fa993** [comment]
> There is just a conflict in the `Cargo.toml`, take both dotenv and the latest version of `clap`

@Mathieu-Lala I updated the version of clap, not sure how to resolve the dotenv conflict

**Comment by Mathieu-Lala** [comment]
You can just remove those lines
```
<<<<<<< develop
=======
>>>>>>> develop
```

**Review by Mathieu-Lala — APPROVED** [review]

File: src/vsmtp/vsmtp-test/src/tests/mod.rs [hunk 1]
```diff
@@ -48,5 +48,6 @@ mod rule_engine {
 mod rules {
     mod codes;
     mod quarantine;
+    mod dotenv;
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-test/src/tests/mod.rs:51]: Format like this:
```rs
  mod dotenv;
  mod quarantine;
```

File: src/vsmtp/vsmtp-core/src/args.rs [hunk 2]
```diff
@@ -89,18 +94,33 @@ mod tests {
...
                 timeout: None
             },
             <Args as clap::Parser>::try_parse_from(["", "-c", "path"]).unwrap()
         );
 
+        assert_eq!(
+            Args {
+                version: false,
+                command: None,
+                config: "path".to_string(),
```
Comment (Mathieu-Lala) [inline src/vsmtp/vsmtp-core/src/args.rs:109]: Fix the test with:
```rs
  config: "/etc/vsmtp/vsmtp.vsl".to_string()
```

**Review by Mathieu-Lala — CHANGES_REQUESTED** [review]
To make the CI pass

**Review by Mathieu-Lala — APPROVED** [review]

**Comment by Mathieu-Lala** [comment]
Good to go, thank you for your contribution to vSMTP ! 🚀

**Comment by fa993** [comment]
Your welcome!

**Pull request merged**